src/util/union_find.cpp

```diff
@@ -64,7 +64,7 @@ void unsigned_union_find::isolate(size_type a)
   // get its root
   size_type r=find(a);
 
-  // assert(r!=a);
+  // assert(r!=a);  // Commented out dead code? To review
```

Comment:
`INVARIANT(r != a, "Non-root nodes must have root that is different from them")`